backend/backend/src/main/java/de/aaaaaaah/velcom/backend/restapi/endpoints/QueueEndpoint.java

```diff
@@ -312,21 +312,37 @@ public GetTaskInfoReply getTask(@PathParam("taskid") UUID taskId) {
 	@GET
 	@Timed(histogram = true)
 	public GetTaskOutputReply getRunnerOutput(@PathParam("taskid") UUID taskId) {
-		Optional<KnownRunner> worker = dispatcher.getKnownRunners().stream()
-			.filter(it -> it.getCurrentTask().isPresent())
-			.filter(it -> it.getCurrentTask().get().getId().getId().equals(taskId))
-			.findAny();
+		Optional<LinesWithOffset> lastOutputLinesOpt = findLinesForTask(taskId);
 
-		if (worker.isEmpty()) {
+		if (lastOutputLinesOpt.isEmpty()) {
 			throw new WebApplicationException(Status.NOT_FOUND);
 		}
 
-		LinesWithOffset lastOutputLines = worker.get().getLastOutputLines()
+		LinesWithOffset lastOutputLines = lastOutputLinesOpt
 			.orElse(new LinesWithOffset(0, List.of()));
 
 		return new GetTaskOutputReply(lastOutputLines.getLines(), lastOutputLines.getFirstLineOffset());
 	}
 
+	private Optional<LinesWithOffset> findLinesForTask(UUID taskId) {
```

Comment:
Maybe move this sort of logic to the dispatcher?

Reply:
Hm, doesn't sound too bad